src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.unit.netcoreapp.cs

```diff
@@ -0,0 +1,460 @@
+            {
+                var watcher = new FileSystemWatcher(testDirectory.Path, "*.pdb");
+                watcher.Filters.Add("foo");
+                Assert.Equal(new string[] { "*.pdb", "foo" }, watcher.Filters);
+            }
+        }
+
+        [Fact]
+        public void FileSystemWatcher_File_Delete_MultipleFilters()
+        {
+            // Check delete events against multiple filters
+
+            DirectoryInfo directory = Directory.CreateDirectory(GetTestFilePath());
+            FileInfo fileOne = new FileInfo(Path.Combine(directory.FullName, GetTestFileName()));
+            FileInfo fileTwo = new FileInfo(Path.Combine(directory.FullName, GetTestFileName()));
+            fileOne.Create().Dispose();
+            fileTwo.Create().Dispose();
+
+            using (var watcher = new FileSystemWatcher(directory.FullName))
+            {
+                watcher.Filters.Add(fileOne.Name);
+                watcher.Filters.Add(fileTwo.Name);
+
+                // using expectedPaths instead of expectedPath because otherwise in this call to ExpectEvent first lambda event handler (added during the last call) will fail.
+                // We cant unregister the labda event handler directly.
+                ExpectEvent(watcher, WatcherChangeTypes.Deleted, () => fileOne.Delete(), cleanup: null, expectedPaths : new string[] { fileOne.FullName, fileTwo.FullName});
+                ExpectEvent(watcher, WatcherChangeTypes.Deleted, () => fileTwo.Delete(), cleanup: null, expectedPaths: new string[] { fileOne.FullName, fileTwo.FullName });
+            }
+        }
+
+        [Fact]
+        public void FileSystemWatcher_Directory_Create_MultipleFilters()
+        {
+            // Check create events against multiple filters
+
+            DirectoryInfo directory = Directory.CreateDirectory(GetTestFilePath());
+            string directoryOne = Path.Combine(directory.FullName, GetTestFileName());
+            string directoryTwo = Path.Combine(directory.FullName, GetTestFileName());
+            string directoryThree = Path.Combine(directory.FullName, GetTestFileName());
+
+            using (var watcher = new FileSystemWatcher(directory.FullName))
+            {
+                watcher.Filters.Add(Path.GetFileName(directoryOne));
+                watcher.Filters.Add(Path.GetFileName(directoryTwo));
+
+                ExpectEvent(watcher, WatcherChangeTypes.Created, () => Directory.CreateDirectory(directoryOne), cleanup: null, expectedPaths: new string[] { directoryOne, directoryTwo, directoryThree });
+                ExpectEvent(watcher, WatcherChangeTypes.Created, () => Directory.CreateDirectory(directoryTwo), cleanup: null, expectedPaths: new string[] { directoryOne, directoryTwo, directoryThree });
+                Assert.Throws<Xunit.Sdk.TrueException>(() => ExpectEvent(watcher, WatcherChangeTypes.Created, () => Directory.CreateDirectory(directoryThree), expectedPaths: new string[] { directoryOne, directoryTwo, directoryThree }));
```

Comment:
This is really awkward. I'd create a helper here, say `ExpectNoEvent()` that does what you're doing here. Especially important as you don't want to keep retrying or have a long timeout in the negative case. Make sure your helper works by creating a failing test temporarily.

Reply:
Additionally, I would be consistent and test the negative for all your tests.